citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -178,15 +183,21 @@ async fn payload_handler(
                 None => info!(target: "citadel","connection not found"),
             };
         }
-        InternalServicePayload::ConnectionFailure { .. } => {}
-        InternalServicePayload::MessageReceived { .. } => {}
-        InternalServicePayload::Disconnect { .. } => {}
+
+        InternalServicePayload::Disconnect { cid, uuid } => match connection_map.get_mut(&cid) {
+            Some(sink) => {
+                let response = InternalServiceResponse::DisconnectSuccess(cid);
```

Comment:
The disconnection request is not a success here. We need to first send a Disconnect `NodeRequest`.